src/components/figures/figure_container/figure_container.ts

```diff
@@ -247,43 +245,64 @@ export class FiguresContainer extends Component<Props, SpreadsheetChildEnv> {
     const { scrollX, scrollY } = this.env.model.getters.getActiveSheetScrollInfo();
     const { x: viewportX, y: viewportY } = this.env.model.getters.getMainViewportCoordinates();
 
-    const left = ["bottomRight", "topRight"].includes(container) ? -(viewportX + scrollX) : 0;
-    const top = ["bottomRight", "bottomLeft"].includes(container) ? -(viewportY + scrollY) : 0;
+    let left = 0;
+    let top = 0;
+
+    if (container === "dnd") {
+      left = -scrollX;
+      top = -scrollY;
+    }
+    if (["bottomRight", "topRight"].includes(container)) {
+      left = -scrollX - viewportX;
+    }
+    if (["bottomRight", "bottomLeft"].includes(container)) {
+      top = -scrollY - viewportY;
+    }
 
     return cssPropertiesToCss({
       left: `${left}px`,
       top: `${top}px`,
     });
   }
 
-  private getFigureContainer(figure: Figure): ContainerType {
+  private getFigureContainer(figureUI: FigureUI): ContainerType {
     const { x: viewportX, y: viewportY } = this.env.model.getters.getMainViewportCoordinates();
-    if (figure.id === this.dnd.draggedFigure?.id) {
+    if (figureUI.id === this.dnd.draggedFigure?.id) {
       return "dnd";
-    } else if (figure.x < viewportX && figure.y < viewportY) {
+    } else if (figureUI.x < viewportX && figureUI.y < viewportY) {
       return "topLeft";
-    } else if (figure.x < viewportX) {
+    } else if (figureUI.x < viewportX) {
       return "bottomLeft";
-    } else if (figure.y < viewportY) {
+    } else if (figureUI.y < viewportY) {
       return "topRight";
     } else {
       return "bottomRight";
     }
   }
 
-  startDraggingFigure(figure: Figure, ev: MouseEvent) {
+  startDraggingFigure(figureUI: FigureUI, ev: MouseEvent) {
     if (ev.button > 0 || this.env.model.getters.isReadonly()) {
       // not main button, probably a context menu and no d&d in readonly mode
       return;
     }
-    const selectResult = this.env.model.dispatch("SELECT_FIGURE", { id: figure.id });
+    const selectResult = this.env.model.dispatch("SELECT_FIGURE", { figureId: figureUI.id });
     if (!selectResult.isSuccessful) {
       return;
     }
 
     const sheetId = this.env.model.getters.getActiveSheetId();
 
     const initialMousePosition = { x: ev.clientX, y: ev.clientY };
+    const initialScrollPosition = this.env.model.getters.getActiveSheetScrollInfo();
+
+    // switch to global viewport position
+    const container = this.getFigureContainer(figureUI);
+    if (["bottomLeft", "topLeft"].includes(container)) {
+      figureUI.x += initialScrollPosition.scrollX;
```

Comment:
I think the previous approach that would create a new object was better because here , you change the values in place but the object is passed as a props, so you actually change a props in place.

Was there a specific reason to get rid of `internalFigureToScreen` and its friends? I found the intention clearer with it (subjective af) but I might miss something related to the new anchor

Reply:
> Was there a specific reason to get rid of internalFigureToScreen and its friends? I found the intention clearer with it (subjective af) but I might miss something related to the new anchor
Here the code only move the figure coordinate if it's part of the frozen pane viewport so that is doesn't teleport to their "true" position, it's not really related to "internal -> screen" old behaviour.
However, i will isolate the resulting code in a method to be more explicit :+1: